src/self-model.scm

```diff
@@ -308,16 +320,66 @@
 ;; Return true if low sound is heard
 (DefineLink
     (DefinedPredicate "very low sound?")
-    (NotLink (GreaterThan
+      (NotLink (GreaterThan
         (Get (State decibel-value (Variable "$y")))
         very-low-sound)))
-
+        
 ;; Return true for normal conversation
 (DefineLink
-    (DefinedPredicate  "normal conversation?")
+   (DefinedPredicate "normal conversation?")
     (NotLink (GreaterThan
         (Get (State decibel-value (Variable "$z")))
         normal-conversation)))
+        
+;; Return true if a very loud sound is heard
+(DefineLink
+	(DefinedPredicate "Heard very loud sound?")
+	(NotLink (GreaterThan
+		(Get (State decibel-value (Variable "$a")))
+		very-loud-sound)))
+;--------------------------------------------
+;;For Saliency
+;;if there are no faces and the degree value is greater than 13, 
+;;then it's considered as salient and The robot is supposed to 
+;;look at the salient position and show curious expression when something salient happens.
+(define-public salient-loc  (AnchorNode "locations"));;obtain the coordinates for the salient location
+(define-public initial-loc (list (NumberNode 1.0)(NumberNode 0)(NumberNode 0)))
+(State salient-loc (List initial-loc))
+
+(define salient (AnchorNode "Degree value"));;obtain the degree of the salient point
+(define initial-degree (Number 0))
+(State salient initial-degree)
+
+
+(DefineLink 
+	(DefinedPredicate "saliency")
+	(GreaterThan
+		(Get (State salient (Variable"$S")))
+		(Number 13)))
+
+(DefineLink
+	(DefinedPredicate "no faces")
+	(EqualLink
+		(DefinedSchemaNode "Num visible faces")
+		(NumberNode 0)))
+
+(DefineLink 
+	(DefinedPredicate "saliency required?")
+	(SequentialAnd
+		(DefinedPredicateNode "no faces")
```

Comment:
could you just copy the EqualLink to here, and remove the "no faces" predicate? That would simplify things slightly.